File: springapp/src/main/java/com/examly/springapp/service/TicketService.java
package com.examly.springapp.service;
 
import java.util.List;
 
import com.examly.springapp.model.Ticket;
 
public interface TicketService {
    Ticket addTicket(Ticket ticket);
    Ticket getTicketById(Long ticketId);
    List<Ticket> getAllTickets();
    Ticket updateTicket(Long ticketId, Ticket ticket);
    Ticket deleteTicket(Long ticketId);
    List<Ticket> getTicketsByAgentId(Long agentId);
    List<Ticket> getTicketsByUserId(Long userId);
}

File: angularapp/src/app/services/ticket.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Ticket } from '../models/ticket.model';

@Injectable({
  providedIn: 'root'
})
export class TicketService {

  public apiUrl='http://localhost:8080';

  constructor(private readonly http:HttpClient) { }

  getAllTickets():Observable<Ticket[]>{
    return this.http.get<Ticket[]>(this.apiUrl);
  }

  getTicketById(ticketId:number):Observable<Ticket>{
    return this.http.get<Ticket>(`${this.apiUrl}/${ticketId}`);
  }

  // https://ide-edfabfeacacebfcdbfeeffbabcfdecbeccfa.premiumproject.examly.io/proxy/8080/

  addTicket(ticket:Ticket):Observable<Ticket>{
    
    return this.http.post<Ticket>(this.apiUrl, ticket);
  }

  updateTicket(ticketId:number,ticket:Ticket):Observable<Ticket>{

    return this.http.put<Ticket>(`${this.apiUrl}/${ticketId}`, ticket);
  }

  deleteTicket(ticketId:number):Observable<void>{
    return this.http.delete<void>(`${this.apiUrl}/${ticketId}`);
  }

  getTicketsByAgentId(agentId:number):Observable<any>{
    return this.http.get<any>(`${this.apiUrl}/agent/${agentId}`);
  }

  getTicketsByUserId(userId:number):Observable<any[]>{
    return this.http.get<any[]>(`${this.apiUrl}/user/${userId}`);
  }
}

